server/src/schema.ts

```diff
@@ -6,190 +6,285 @@ import utils from './utils'
-        throw new Error('Error. No token')
-      }
-      let user = await ctx.prisma.user.findFirst({
-        where: {
-          resetPasswordToken: args.resetPasswordToken,
-        },
-      })
-
-      if (!user) {
-        throw new Error('Link not valide!')
+          return {
+            feedId,
+          }
+        }
+      } catch (error) {
+        console.error(error)
+        throw new UserInputError(error.message || ':(')
       }
+    },
 
-      if (!user.dateResetPasswordRequest) {
-        throw new Error('User never requested a link')
-      }
-      if (!user.resetPasswordToken) {
-        throw new Error('User never requested a link.')
-      }
+    updateUserSettings: async (parent, args, ctx: Context) => {
+      const { userId, theme } = args
 
-      let t = new Date(user.dateResetPasswordRequest)
-      t.setSeconds(t.getSeconds() + 360)
+      try {
+        const setting = await ctx.prisma.setting.findFirst({
+          where: {
+            userId,
+          },
+        })
 
-      if (new Date(t).getTime() < new Date().getTime()) {
-        throw new Error('Link is out of date ')
-      }
-      user = await ctx.prisma.user.update({
-        where: {
-          id: user.id,
-        },
-        data: {
-          resetPasswordToken: '',
-          dateResetPasswordRequest: null,
-        },
-      })
-      return {
-        user,
-        token: jwt.sign({ userId: user.id }, config.APP_SECRET, {
-          expiresIn: '2d',
-        }),
-      }
-    },
-    signupUser: async (parent, args, ctx: Context) => {
-      const userTest = await ctx.prisma.user.findUnique({
-        where: {
-          email: args.email,
-        },
-      })
-      if (userTest) {
-        throw new Error('User Already exists')
-      }
-      utils.isPasswordSafe(args.password)
-      const password = await bcrypt.hash(args.password, 10)
+        if (!setting) {
+          throw new UserInputError('Setting not found')
+        }
 
-      const validateEmailToken = crypto.randomBytes(64).toString('hex')
-      const user = await ctx.prisma.user.create({
-        data: {
-          name: args.name,
-          password: password,
-          role: 'USER',
-          email: args.email,
-          resetPasswordToken: '',
-          lastLogin: new Date(),
-          validateEmailToken,
-          isEmailValidated: false,
-        },
-      })
-      return {
-        user,
-        token: jwt.sign({ userId: user.id }, config.APP_SECRET, {
-          expiresIn: '2d',
-        }),
-      }
-    },
-    loginUser: async (parent, args, ctx: Context) => {
-      let user = await ctx.prisma.user.findFirst({
-        where: {
-          email: args.email,
-        },
-      })
-      if (!user) {
-        throw new Error('No user')
-      }
+        const newSetting = await ctx.prisma.setting.update({
+          where: {
+            userId,
+          },
+          data: {
+            ...(theme && { theme }),
```

Comment:
super unreadable. i can understand what it does but clean code doesn't involve using cool tricks, to be honest.
for example, if you have multiple settings
```
const newSettings = {}
if (theme) {
   newSettings.theme = theme;
}
```

or, by the way, I think just doing:
```
data: {
   theme: theme,
}
```

works here (or something like this)